### PR TITLE
Setup linters

### LINKED ISSUE #4: Set up linters
- Set up eslint.
- Set up Stylelint.
- Set up node checker

### PR DESCRIPTION
**Pull request by yonas44**
### Summary
- Set up github actions.
- Set up `eslint` and `stylelint` to lint locally.
- I moved the `eslintrc.cjs` to the root directory to lint all `.ts` files

### CODE REVIEW

**Comment by Abdessittir** [comment]
Hi, @yonas44, thanks for adding the linters, I will review this PR 
- [ ] Kindly link this PR to its corresponding issue

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "dev:cl": "cd frontend && npm run start",
     "dev:ts": "tsc -w",
     "dev:js": "nodemon dist/server.js",
-    "dev": "concurrently npm:dev:*"
+    "dev": "concurrently npm:dev:*",
+    "lint": "npx eslint --fix . && npx stylelint \"**/*.{css,scss}\""
```
Comment (AHP15) [inline package.json]: - [ ] I have question, I can see that added the `--fix` flag to fix eslint errors, but how about stylelint, should we add the same flag here after `npx stylelint \"**/*.{css,scss}\"` ?

**Review by AHP15 — CHANGES_REQUESTED** [review]
Hi @yonas44, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be great if ignore the `dist` folders because when I'm running it locally it shows a lot of error


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: package.json (hunk 1 above)
Comment (yonas44) [reply package.json]: Thanks for the feedback, I will add the --fix flag and add the dist directory in the .gitignore 👍

Comment (AHP15) [reply package.json]: Hi @yonas44, This dist folder is already in .gitignore.
I'm talking about is there any way to ignore when we run the command `npm run lint`

Comment (yonas44) [reply package.json]: I am thinking of adding an ignore-pattern in the linter config file to skip specific directories

**Review by AHP15 — APPROVED** [review]
Hi @yonas44 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

:heavy_check_mark: Docker is running locally 
:heavy_check_mark: all tests are passing
:heavy_check_mark: No linter errors

Cheers, and Happy coding!👏👏👏

**Pull request merged**